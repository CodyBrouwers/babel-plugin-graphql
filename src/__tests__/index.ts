import path from "path";
import { transformSync } from "@babel/core";
import pluginTester from "babel-plugin-tester";
import plugin from "..";

// const projectRoot = path.join(__dirname, "../../");

// expect.addSnapshotSerializer({
//   print(val) {
//     return val.split(projectRoot).join("<PROJECT_ROOT>/");
//   },
//   test(val) {
//     return typeof val === "string";
//   },
// });

// const fixture = (filename: string) => ({
//   fixture: require.resolve(`./fixtures/${filename}`),
// });

// pluginTester({
//   plugin,
//   snapshot: false,
//   babelOptions: { filename: __filename },
//   tests: {
//     "basic test of functionality": fixture("basic"),
//     // "exportable query": fixture("exportconstquery"),
//     // "basic array map works": fixture("arraymap1"),
//     // "array map destructuring arrow function works": fixture("arraymap2"),
//     // "array map destructuring normal function works": fixture("arraymap3"),
//     // "assignments and aliases work": fixture("assignment"),
//     // "overlapping assignments work": fixture("assignment-overlapping"),
//     // "destructuring work": fixture("destructuring"),
//     // "docs: fields": fixture("fields"),
//     // "docs: args": fixture("arguments"),
//     // "docs: variables": fixture("queryVariables"),
//     // "docs: directives": fixture("directives"),
//     // "injection of fragments": fixture("fragment"),
//   },
// });

it("handles nested properties", () => {
  const input = `
    function Movie() {
      const { data } = useQuery("GetMovie");

      return (
        <div>
          <p>{data.Movie.releaseDate}</p>
          <p>{data.Movie.director.id}</p>
          <p>{data.Movie.director.name({first: 5})}</p>
          <p>{data.Movie.director.name}</p>
        </div>
      );
    }
    `;
  const { code } = transformSync(input, { filename: "./fixtures/basic.js", plugins: [plugin] });
  console.log(code);
  expect(code).toBeTruthy();
});

// it("handles arguments", () => {
//   var input = `
//     function Movie() {
//       const { data } = useQuery("GetMovie");

//       return (
//         <div>
//           <p>{data.Movie.releaseDate({numberArg: 123, stringArg: "abc"}, "@preload")}</p>
//         </div>
//       );
//     }
//     `;
//   const { code } = transformSync(input, { filename: "./fixtures/basic.js", plugins: [plugin] });
//   console.log(code);
//   expect(code).toBeTruthy();
// });
