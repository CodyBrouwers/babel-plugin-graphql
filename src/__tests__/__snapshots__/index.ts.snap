// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-blade basic test of functionality: basic test of functionality 1`] = `

function Movie() {
  const { data } = useQuery("Movie");

  return (
    <div>
      <div>
        <h2>{data.Movie.movie.gorilla}</h2>
        <p>{data.Movie.movie.monkey}</p>
        <p>{data.Movie.chimp}</p>
      </div>
    </div>
  );
}

      ↓ ↓ ↓ ↓ ↓ ↓

function Movie() {
  const {
    data
  } = useQuery("Movie");
  return <div>
      <div>
        <h2>{data.Movie.movie.gorilla}</h2>
        <p>{data.Movie.movie.monkey}</p>
        <p>{data.Movie.chimp}</p>
      </div>
    </div>;
}

`;

exports[`babel-plugin-graphql basic test of functionality: basic test of functionality 1`] = `

import Thing from "./thing";

function Movie() {
  const { data } = useQuery("GetMovie");

  return (
    <div>
      <div>
        <p>{data.aaa.bbb.ccc.ddd}</p>
      </div>
    </div>
  );
}

// <p>{data.id}</p>
// <p>{data.director.id}</p>
// <p>{data.nodd.banana}</p>
// <h2>{data.name}</h2>
// <h2>{data.id}</h2>
// <p>{data.releaseDate.formatted}</p>

      ↓ ↓ ↓ ↓ ↓ ↓

import Thing from "./thing";

function Movie() {
  const { data } = useQuery("GetMovie");
  return (
    <div>
      <div>
        <p>{data.aaa.bbb.ccc.ddd}</p>
      </div>
    </div>
  );
} // <p>{data.id}</p>
// <p>{data.director.id}</p>
// <p>{data.nodd.banana}</p>
// <h2>{data.name}</h2>
// <h2>{data.id}</h2>
// <p>{data.releaseDate.formatted}</p>


`;
